### PR TITLE
Refactor examples_test.walkJSONFiles

### PR DESCRIPTION
**Pull request by tmrts**
While I was walking through the examples, I found the flow of this function to be confusing, so I decided to make it a bit more clearer. I hope it's better now.

### CODE REVIEW

File: examples/examples_test.go [hunk 1]
```diff
@@ -70,30 +70,30 @@ func validateObject(obj runtime.Object) (errors []error) {
...
 			return filepath.SkipDir
 		}
-		name := filepath.Base(path)
-		ext := filepath.Ext(name)
-		if ext != "" {
-			name = name[:len(name)-len(ext)]
-		}
-		if !(ext == ".json" || ext == ".yaml") {
+
+		dir, file := filepath.Split(path)
```
Comment (vmarmol) [inline examples/examples_test.go]: Looks like dir is  unused, Go usually notices those am I missing something?

Comment (vmarmol) [inline examples/examples_test.go]: If not, can be "_"

**Comment by vmarmol** [comment]
Looks good overall, just the one small comment. Could you squash your commits once the fix is in? Thanks for making the example more readable!

**Comment by TamerTas** [comment]
No problem at all. Let me squash the commits.

**Pull request closed without merging**